src/lib/types/option.ts

```diff
@@ -53,10 +55,16 @@ export enum IAuthType {
     NONE = 'none',
 }
 
+export type CustomHandlerType = (
+    app: Express,
+    config: IUnleashConfig,
+    services: IUnleashServices,
+) => void;
+
 export interface IAuthOption {
     enableApiToken: boolean;
     type: IAuthType;
-    customAuthHandler?: Function;
+    customAuthHandler?: CustomHandlerType;
```

Comment:
Finally, this loosely typed function is what caused us to miss the removal of the config parameter in this incident: https://unleash.instatus.com/clofjecqc147784bpogyb47nkpz

The type of the function is now tied to the parameter types we send in when we use it: https://github.com/Unleash/unleash/blob/312999066b0f712eecfeebd10f418600a0862c04/src/lib/app.ts#L121